actors/builtin/miner/miner_actor.go

```diff
@@ -1498,6 +1498,13 @@ func (a Actor) ReportConsensusFault(rt Runtime, params *ReportConsensusFaultPara
 	burnAmount := big.Zero()
 	rewardAmount := big.Zero()
 	rt.State().Transaction(&st, func() {
+		info := getMinerInfo(rt, &st)
+
+		// verify miner hasn't already been faulted
+		if fault.Epoch < info.ConsensusFaultElapsed {
+			rt.Abortf(exitcode.ErrForbidden, "miner has existing fault that has not expired (this may be a duplicate)")
```

Comment:
Nit: the message is still misleading. `ConsensusFaultElapsed` may have expired, but the fault is older.

`fault epoch %d is too old, last exclusion period ended at %d`